### DISCUSSION TITLE
Buffer invalidation in FAST - DDS

### DISCUSSION THREAD

**Discussion by AnoopEmbeddedEngineer** [eProsima/Fast-DDS#5363] [General]
When i tried setting maxmessagesize equal to segmentsize in SHM transport of fast dds...
It is showing buffer invalidation warning...But i am not getting any data loss....What is buffer invalidation in fast dds...

I want to know, even if buffer invalidation is happening, why there is no data loss?
![Screenshot from 2024-10-28 23-16-52](https://github.com/user-attachments/assets/413dcedc-0f35-4bc8-abfe-8fd13f0570a1)

**Comment by paxifaer** [comment]
Can  you show us your xml setting?

**Reply by AnoopEmbeddedEngineer** [reply]
> Can you show us your xml setting?

<profiles xmlns="http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles">
    <transport_descriptors>
        
        <transport_descriptor>
            <transport_id>shm_transport</transport_id>
            <type>SHM</type>
	    <maxMessageSize>134217728</maxMessageSize>
        </transport_descriptor>
    </transport_descriptors>

    <participant profile_name="SHMParticipant" is_default_profile="true">
        <rtps>
            
            <userTransports>
                <transport_id>shm_transport</transport_id>
            </userTransports>
            <useBuiltinTransports>false</useBuiltinTransports>
        </rtps>
    </participant>
</profiles>

**Reply by paxifaer** [reply]
Maybe your computer does not allocate enough shared memory.
